### ISSUE TITLE
Unable to deploy created classes

### ISSUE THREAD

**Issue by javydreamercsw** [jGauravGupta/JPAModeler#10]
I made a design in the modeler and created the classes. When I try to create the entities in the database I'm getting the following error:

```
Exception [EclipseLink-30005] (Eclipse Persistence Services - 2.6.2.v20151217-774c696): org.eclipse.persistence.exceptions.PersistenceUnitLoadingException
Exception Description: An exception was thrown while searching for persistence archives with ClassLoader: sun.misc.Launcher$AppClassLoader@73d16e93
Internal Exception: javax.persistence.PersistenceException: Exception [EclipseLink-28018] (Eclipse Persistence Services - 2.6.2.v20151217-774c696): org.eclipse.persistence.exceptions.EntityManagerSetupException
Exception Description: Predeployment of PersistenceUnit [QT_PU] failed.
Internal Exception: java.lang.ClassCastException: org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.IdAccessor cannot be cast to org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.ObjectAccessor
	at org.eclipse.persistence.exceptions.PersistenceUnitLoadingException.exceptionSearchingForPersistenceResources(PersistenceUnitLoadingException.java:127)
	at org.eclipse.persistence.jpa.PersistenceProvider.createEntityManagerFactoryImpl(PersistenceProvider.java:115)
	at org.eclipse.persistence.jpa.PersistenceProvider.createEntityManagerFactory(PersistenceProvider.java:188)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:79)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:54)
```

Something is off, but can't figure it out. I'll create the same model by NetBeans default means and compare. I'll update with any findings.

**Comment by javydreamercsw** [comment]
Adding diagram file for investigation.
[Diagram.jpa.zip](https://github.com/jGauravGupta/JPAModeler/files/240912/Diagram.jpa.zip)

**Comment by javydreamercsw** [comment]
Message : org.netbeans.jpa.modeler.db.accessor.IdSpecAccessor cannot be cast to org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.ObjectAccessor

StackTrace : 
java.lang.ClassCastException: org.netbeans.jpa.modeler.db.accessor.IdSpecAccessor cannot be cast to org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.ObjectAccessor
	at org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.ObjectAccessor.processId(ObjectAccessor.java:489)
	at org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.ObjectAccessor.processOwningMappingKeys(ObjectAccessor.java:700)
	at org.eclipse.persistence.internal.jpa.metadata.accessors.mappings.ManyToOneAccessor.process(ManyToOneAccessor.java:123)
	at org.netbeans.jpa.modeler.db.accessor.ManyToOneSpecAccessor.process(ManyToOneSpecAccessor.java:53)
	at org.eclipse.persistence.internal.jpa.metadata.accessors.classes.ClassAccessor.processDerivedId(ClassAccessor.java:1619)
	at org.eclipse.persistence.internal.jpa.metadata.accessors.classes.EntityAccessor.processDerivedId(EntityAccessor.java:993)
	at org.eclipse.persistence.internal.jpa.metadata.MetadataProject.processAccessorsWithDerivedIDs(MetadataProject.java:1539)
	at org.eclipse.persistence.internal.jpa.metadata.MetadataProject.processStage3(MetadataProject.java:1836)
	at org.eclipse.persistence.internal.jpa.metadata.MetadataProcessor.processORMMetadata(MetadataProcessor.java:577)
	at org.netbeans.db.modeler.persistence.internal.jpa.deployment.JPAMPersistenceUnitProcessor.processORMetadata(JPAMPersistenceUnitProcessor.java:49)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.createDBMapping(DBModelerUtil.java:211)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.loadModelerFile(DBModelerUtil.java:148)
	at org.netbeans.modeler.core.NBModelerUtil.loadModelerFile(NBModelerUtil.java:57)
	at org.netbeans.modeler.specification.model.file.action.ModelerFileActionListener.openModelerFile(ModelerFileActionListener.java:155)
	at org.netbeans.db.modeler.specification.model.file.action.DBViewerActionListener.init(DBViewerActionListener.java:61)
	at org.netbeans.jpa.modeler.specification.model.util.JPAModelerUtil.openDBViewer(JPAModelerUtil.java:1645)
	at org.netbeans.jpa.modeler.specification.model.scene.JPAModelerScene.lambda$getPopupMenuItemList$6(JPAModelerScene.java:353)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at org.netbeans.core.TimableEventQueue.dispatchEvent(TimableEventQueue.java:159)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


ModelerFile : 
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<jpa:entity-mappings xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:java="http://jcp.org/en/jsr/detail?id=270" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:jpa="http://java.sun.com/xml/ns/persistence/orm" pkg="net.sourceforge.javydreamercsw.database.db" v="1.5.4" thm="PMacColorScheme" persistenceUnitName="QT_PU" xs="false" id="_146162083179159" xsi:schemaLocation="http://java.sun.com/xml/ns/persistence/orm orm_2_1.xsd">
    <jpa:default-class abs="false" class="ComplaintPK" visibile="true" minimized="false">
        <jpa:embeddable>false</jpa:embeddable>
        <jpa:attribute attribute-type="Long" visibile="true" name="id"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="country"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="investigation"/>
    </jpa:default-class>
    <jpa:default-class abs="false" class="InvestigationPK" visibile="true" minimized="false">
        <jpa:embeddable>false</jpa:embeddable>
        <jpa:attribute attribute-type="Long" visibile="true" name="id"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="failureMode"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="rootCause"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="cause"/>
    </jpa:default-class>
    <jpa:entity xre="false" sm="false" abs="false" class="Product" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083180260">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="name" id="_146162083180261"/>
            <basic attribute-type="String" visibile="true" name="part_number" id="_146162083180363">
                <jpa:column unique="true" nullable="true" insertable="true" updatable="true" length="255" precision="0" scale="0"/>
            </basic>
            <one-to-many own="true" collection-type="java.util.List" connected-entity-id="_146162083181065" visibile="true" name="complaints" id="_146162083180362"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083180364"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" compositePrimaryKeyType="IDCLASS" compositePrimaryKeyClass="ComplaintPK" sm="false" abs="false" class="Complaint" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083181065">
        <jpa:attributes>
            <basic attribute-type="Float" visibile="true" name="quantity" id="_146162083181067"/>
            <basic attribute-type="String" visibile="true" name="description" id="_146162083181068"/>
            <basic attribute-type="java.sql.Date" visibile="true" name="report_date" id="_146162083181171"/>
            <basic attribute-type="String" visibile="true" name="exception_number" id="_1461679173354119"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083188889" connected-attribute-id="_146162083189091" visibile="true" name="country" id="_146162083181066"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083183076" connected-attribute-id="_146162083183080" visibile="true" name="investigation" id="_146162083181169"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083181170"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity name="Failure Mode" xre="false" sm="false" abs="false" class="FailureMode" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083182072">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="mode" id="_146162083182073">
                <jpa:column unique="true" nullable="true" insertable="true" updatable="true" length="255" precision="0" scale="0"/>
            </basic>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083183076" connected-attribute-id="_146162083183078" visibile="true" name="investigations" id="_146162083182074"/>
            <one-to-many own="true" collection-type="java.util.List" connected-entity-id="_146162083182072" visibile="true" name="failureModes" id="_146185579989414"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083182075"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" compositePrimaryKeyType="IDCLASS" compositePrimaryKeyClass="InvestigationPK" sm="false" abs="false" class="Investigation" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083183076">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="result" id="_146162083183077"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083182072" connected-attribute-id="_146162083182074" visibile="true" name="failureMode" id="_146162083183078"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083184083" visibile="true" name="rootCause" id="_146162083183079"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083186685" connected-attribute-id="_146162083186687" visibile="true" name="cause" id="_146162083183081"/>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083181065" connected-attribute-id="_146162083181169" visibile="true" name="complaints" id="_146162083183080"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083183082"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity name="Root Cause" xre="false" sm="false" abs="false" class="RootCause" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083184083">
        <jpa:attributes>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083184084"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" sm="false" abs="false" class="Cause" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083186685">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="cause" id="_146162083186686">
                <jpa:column unique="true" nullable="true" insertable="true" updatable="true" length="255" precision="0" scale="0"/>
            </basic>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083183076" connected-attribute-id="_146162083183081" visibile="true" name="investigations" id="_146162083186687"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083186688"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" sm="false" abs="false" class="Country" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083188889">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="name" id="_146162083188890"/>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083181065" connected-attribute-id="_146162083181066" visibile="true" name="complaints" id="_146162083189091"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083189092"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:diagram>
        <plane elementRef="_146162083179159">
            <shape elementRef="_146162083180260">
                <Bounds x="280.0" y="32.0" width="151.0" height="138.0"/>
            </shape>
            <shape elementRef="_146162083181065">
                <Bounds x="533.0" y="234.0" width="167.0" height="198.0"/>
            </shape>
            <shape elementRef="_146162083182072">
                <Bounds x="32.0" y="234.0" width="184.0" height="138.0"/>
            </shape>
            <shape elementRef="_146162083183076">
                <Bounds x="280.0" y="476.0" width="189.0" height="178.0"/>
            </shape>
            <shape elementRef="_146162083184083">
                <Bounds x="533.0" y="718.0" width="173.0" height="54.0"/>
            </shape>
            <shape elementRef="_146162083186685">
                <Bounds x="32.0" y="718.0" width="144.0" height="118.0"/>
            </shape>
            <shape elementRef="_146162083188889">
                <Bounds x="770.0" y="476.0" width="152.0" height="118.0"/>
            </shape>
        </plane>
    </jpa:diagram>
</jpa:entity-mappings>
```

**Comment by javydreamercsw** [comment]
This might be related. As I try to Micro DB tables, some work and others don't. Having issues with Investigation table.

Message : Not supported yet.

StackTrace : 
java.lang.UnsupportedOperationException: Not supported yet.
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.getEdgeTargetPinWidget(DBModelerUtil.java:584)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.loadEdge(DBModelerUtil.java:339)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.lambda$null$3(DBModelerUtil.java:315)
	at java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1612)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.lambda$loadFlowEdge$4(DBModelerUtil.java:314)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.loadFlowEdge(DBModelerUtil.java:311)
	at org.netbeans.db.modeler.specification.model.util.DBModelerUtil.loadModelerFile(DBModelerUtil.java:155)
	at org.netbeans.modeler.core.NBModelerUtil.loadModelerFile(NBModelerUtil.java:57)
	at org.netbeans.modeler.specification.model.file.action.ModelerFileActionListener.openModelerFile(ModelerFileActionListener.java:155)
	at org.netbeans.db.modeler.specification.model.file.action.DBViewerActionListener.init(DBViewerActionListener.java:61)
	at org.netbeans.jpa.modeler.specification.model.util.JPAModelerUtil.openDBViewer(JPAModelerUtil.java:1645)
	at org.netbeans.jpa.modeler.core.widget.EntityWidget.lambda$getPopupMenuItemList$4(EntityWidget.java:246)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at org.netbeans.core.TimableEventQueue.dispatchEvent(TimableEventQueue.java:159)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


ModelerFile : 
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<jpa:entity-mappings xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:java="http://jcp.org/en/jsr/detail?id=270" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:jpa="http://java.sun.com/xml/ns/persistence/orm" pkg="net.sourceforge.javydreamercsw.database.db" v="1.5.4" thm="PMacColorScheme" persistenceUnitName="QT_PU" xs="false" id="_146162083179159" xsi:schemaLocation="http://java.sun.com/xml/ns/persistence/orm orm_2_1.xsd">
    <jpa:default-class abs="false" class="ComplaintPK" visibile="true" minimized="false">
        <jpa:embeddable>false</jpa:embeddable>
        <jpa:attribute attribute-type="Long" visibile="true" name="id"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="country"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="investigation"/>
    </jpa:default-class>
    <jpa:default-class abs="false" class="InvestigationPK" visibile="true" minimized="false">
        <jpa:embeddable>false</jpa:embeddable>
        <jpa:attribute attribute-type="Long" visibile="true" name="id"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="failureMode"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="rootCause"/>
        <jpa:attribute attribute-type="Long" visibile="true" name="cause"/>
    </jpa:default-class>
    <jpa:entity xre="false" sm="false" abs="false" class="Product" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083180260">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="name" id="_146162083180261"/>
            <basic attribute-type="String" visibile="true" name="part_number" id="_146162083180363">
                <jpa:column unique="true" nullable="true" insertable="true" updatable="true" length="255" precision="0" scale="0"/>
            </basic>
            <one-to-many own="true" collection-type="java.util.List" connected-entity-id="_146162083181065" visibile="true" name="complaints" id="_146162083180362"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083180364"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" compositePrimaryKeyType="IDCLASS" compositePrimaryKeyClass="ComplaintPK" sm="false" abs="false" class="Complaint" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083181065">
        <jpa:attributes>
            <basic attribute-type="Float" visibile="true" name="quantity" id="_146162083181067"/>
            <basic attribute-type="String" visibile="true" name="description" id="_146162083181068"/>
            <basic attribute-type="java.sql.Date" visibile="true" name="report_date" id="_146162083181171"/>
            <basic attribute-type="String" visibile="true" name="exception_number" id="_1461679173354119"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083188889" connected-attribute-id="_146162083189091" visibile="true" name="country" id="_146162083181066"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083183076" connected-attribute-id="_146162083183080" visibile="true" name="investigation" id="_146162083181169"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083181170"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity name="Failure Mode" xre="false" sm="false" abs="false" class="FailureMode" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083182072">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="mode" id="_146162083182073">
                <jpa:column unique="true" nullable="true" insertable="true" updatable="true" length="255" precision="0" scale="0"/>
            </basic>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083183076" connected-attribute-id="_146162083183078" visibile="true" name="investigations" id="_146162083182074"/>
            <one-to-many own="true" collection-type="java.util.List" connected-entity-id="_146162083182072" visibile="true" name="failureModes" id="_146185579989414"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083182075"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" compositePrimaryKeyType="IDCLASS" compositePrimaryKeyClass="InvestigationPK" sm="false" abs="false" class="Investigation" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083183076">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="result" id="_146162083183077"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083182072" connected-attribute-id="_146162083182074" visibile="true" name="failureMode" id="_146162083183078"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083184083" visibile="true" name="rootCause" id="_146162083183079"/>
            <many-to-one optional="true" primaryKey="true" connected-entity-id="_146162083186685" connected-attribute-id="_146162083186687" visibile="true" name="cause" id="_146162083183081"/>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083181065" connected-attribute-id="_146162083181169" visibile="true" name="complaints" id="_146162083183080"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083183082"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity name="Root Cause" xre="false" sm="false" abs="false" class="RootCause" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083184083">
        <jpa:attributes>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083184084"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" sm="false" abs="false" class="Cause" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083186685">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="cause" id="_146162083186686">
                <jpa:column unique="true" nullable="true" insertable="true" updatable="true" length="255" precision="0" scale="0"/>
            </basic>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083183076" connected-attribute-id="_146162083183081" visibile="true" name="investigations" id="_146162083186687"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083186688"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:entity xre="false" sm="false" abs="false" class="Country" visibile="true" minimized="false" rootElement="_146162083179159" id="_146162083188889">
        <jpa:attributes>
            <basic attribute-type="String" visibile="true" name="name" id="_146162083188890"/>
            <one-to-many own="false" collection-type="java.util.List" connected-entity-id="_146162083181065" connected-attribute-id="_146162083181066" visibile="true" name="complaints" id="_146162083189091"/>
            <jpa:id attribute-type="Long" visibile="true" name="id" id="_146162083189092"/>
        </jpa:attributes>
    </jpa:entity>
    <jpa:diagram>
        <plane elementRef="_146162083179159">
            <shape elementRef="_146162083180260">
                <Bounds x="280.0" y="32.0" width="151.0" height="138.0"/>
            </shape>
            <shape elementRef="_146162083181065">
                <Bounds x="533.0" y="234.0" width="167.0" height="198.0"/>
            </shape>
            <shape elementRef="_146162083182072">
                <Bounds x="32.0" y="234.0" width="184.0" height="138.0"/>
            </shape>
            <shape elementRef="_146162083183076">
                <Bounds x="280.0" y="476.0" width="189.0" height="178.0"/>
            </shape>
            <shape elementRef="_146162083184083">
                <Bounds x="533.0" y="718.0" width="173.0" height="54.0"/>
            </shape>
            <shape elementRef="_146162083186685">
                <Bounds x="32.0" y="718.0" width="144.0" height="118.0"/>
            </shape>
            <shape elementRef="_146162083188889">
                <Bounds x="770.0" y="476.0" width="152.0" height="118.0"/>
            </shape>
        </plane>
    </jpa:diagram>
</jpa:entity-mappings>
```

**Comment by jGauravGupta** [comment]
Hi @javydreamercsw ,

DB Modeler have some bug for complex relationship scenario, that will fixed soon (It is in beta state).

Thanks for your support :)

**Comment by javydreamercsw** [comment]
Sorry to hear that. Besides that I love the tool and is looking promising. Sadly I'll have to switch back to my original and painful method.

MySQL Designer -> MySQL -> Netbeans

Looking forward to when I can skip MySQL. Keep up the great work!

**Comment by jGauravGupta** [comment]
Only DB Modeler Component is in beta state except that all of the component like JPA Modeler designer , Java Reverse Engineering module, DB Reverse Engineering module and JEE8 generator can be used for production code.

DB Modeler is the powerful and complex component of JPA Modeler.

**Comment by jGauravGupta** [comment]
And one more important thing, We are trying to cover mostly all of the basic and complex scenario (based on the developer usage and priority).
So we need your support, just report the bug and we will fix it and provide the support for such scenario ASAP.

Thanks again!

**Comment by javydreamercsw** [comment]
I made a reverse engineering from a working DB created with MySQL Designer. Here's the resulting diagram if of any interest.
[Diagram.jpa.zip](https://github.com/jGauravGupta/JPAModeler/files/240991/Diagram.jpa.zip)

**Comment by javydreamercsw** [comment]
And the related working Entity classes.
[Entities.zip](https://github.com/jGauravGupta/JPAModeler/files/240996/Entities.zip)

**Comment by jGauravGupta** [comment]
Log(Message + stacktrace + xml) copied/generated from tools are enough(that you have already submitted).

**Comment by jGauravGupta** [comment]
Issue has been resolved, please confirm.
Download : https://github.com/jGauravGupta/JPAModeler/releases/tag/1.5.6

In the next release we will cover all derived identity case scenario(as specified in JPA spec) for DB Modeler and will be released ASAP.

**Comment by javydreamercsw** [comment]
It looks great, thanks for the great work!

**Comment by javydreamercsw** [comment]
I had to disable the plugin as the IDE and laptop became unresponsive while
I had the IDE open. After disabling it everything is back to normal.

**Comment by jGauravGupta** [comment]
Please raise new issue and attach the JPA Modeler file.

**Comment by jGauravGupta** [comment]
Issue has been fixed, Please install JPA Modeler v1.6 [https://github.com/jGauravGupta/JPAModeler/releases/tag/1.6](https://github.com/jGauravGupta/JPAModeler/releases/tag/1.6)
